code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -487,56 +497,102 @@
 
 	inventory_head = new/obj/item/clothing/head/christmas/santahat/red(src)
 	regenerate_icons()
+	
+/mob/living/simple_animal/corgi/Ian/Destroy()
+	..()
+	master = null
+	pointer_caller = null
 
 /mob/living/simple_animal/corgi/Ian/Life()
 	if(timestopped)
 		return 0 //under effects of time magick
 
 	..()
-
-	//Feeding, chasing food, FOOOOODDDD
-	if(!stat && !resting && !locked_to && (ckey == null))
-		turns_since_scan++
-		if(turns_since_scan > 5)
-			turns_since_scan = 0
-			if((movement_target) && !(isturf(movement_target.loc) || ishuman(movement_target.loc) ))
+	
+	if(!stat && !resting && !locked_to && (ckey == null)) //Behavior mechanisms (om nom :3)
+		if(ian_status == IDLE)
+			get_target()
+			stop_automated_movement = 0		
+			
+		else if(ian_status == BEGIN_FOOD_HUNTING)
+			ian_status = FOOD_HUNTING
+			spawn(0) // Separate process
+				stop_automated_movement = 1
+				var/failedsteps = 0
+				while(failedsteps <= 3)
```

Comment:
So you just want to have it step 3 times per life?

Please dont do this. Just hold reference to the target and step once per life loop

Reply:
You're misreading the code, after it fails to move 3 times it will stop the loop. It only triggers once per life tick.